### PR TITLE
FIP-82 Upgrade governance

### PR DESCRIPTION
**Pull request by thomas-waite**
# Summary
Introduces the `FIP-82 Governance Upgrade`. 

The cornerstone is the introduction of optimistic pods (a Gnosis Safe with a membership wrapper and an optimistic timelock) to govern limited sections of the protocol. A TribalCouncil is introduced to manage these pods and becomes the de-facto entity which will govern the protocol on an operational basis. The TribeDAO remains the ultimate authority in the Tribe ecosystem.

A NopeDAO is also introduced. This is able to veto any pod proposal, but not the TribeDAO. It has a low quorum threshold of 10 million TRIBE to lower the barrier to vetoing malformed proposals. It's parameters are governable by the Tribe DAO.

### CODE REVIEW

File: contracts/pods/GovernanceMetadataRegistry.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        bytes32 proposalHash = keccak256(
+            abi.encode(podId, proposalId, metadata)
+        );
+        require(
+            registration[proposalHash] == false,
+            "Proposal already registered"
+        );
+
+        emit RegisterProposal(podId, proposalId, metadata);
```
Comment (ElliotFriedman) [inline contracts/pods/GovernanceMetadataRegistry.sol:59]: nit: I would put the event at the end of the function to keep style consistent

**Comment by thomas-waite** [comment]
Removing ZERO_ADDRESS checks: https://forum.openzeppelin.com/t/removing-address-0x0-checks-from-openzeppelin-contracts/2222

File: protocol-configuration/permissions.ts [hunk 2]
```diff
@@ -29,9 +29,14 @@ export const permissions = {
   BALANCER_MANAGER_ADMIN_ROLE: [],
   PSM_ADMIN_ROLE: [],
   TRIBAL_CHIEF_ADMIN_ROLE: ['optimisticTimelock', 'tribalChiefSyncV2'],
```
Comment (thomas-waite) [inline protocol-configuration/permissions.ts:31]: TODO: This `permissions.ts` will become out of date when the `TribalCouncil` starts creating and granting roles

File: contracts/pods/RoleBastion.sol [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    /// @param role Role to be created
+    /// @dev Function is used by an address with ROLE_ADMIN, e.g. the TribalCouncil, to create
+    ///      non-major roles. A check is made to ensure that the role to be created does not
+    ///      have an admin i.e. that it is a new admin.
+    ///      Specificially, can not create the bytes32(0) role as this is set as the CONTRACT_ADMIN_ROLE
+    ///      in core and has not been created yet.
+    function createRole(bytes32 role)
+        external
+        onlyTribeRole(TribeRoles.ROLE_ADMIN)
+    {
```
Comment (ElliotFriedman) [inline contracts/pods/RoleBastion.sol:25]: Add pausing to this function

File: contracts/pods/PodFactory.sol [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+        external
+        override
+        returns (
+            uint256[] memory,
+            address[] memory,
+            address[] memory
+        )
+    {
+        require(
+            burnerDeploymentUsed == false,
```
Comment (ElliotFriedman) [inline contracts/pods/PodFactory.sol:163]: recommend using openzeppelin Initializable instead

File: contracts/dao/nopeDAO/NopeDAO.sol [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    uint256 private _quorum = 10_000_000e18;
+
+    /// @notice Additional governance events
+    event QuorumUpdated(uint256 oldQuorum, uint256 newQuorum);
+
+    constructor(ERC20VotesComp _tribe, address _core)
+        Governor("NopeDAO")
+        GovernorSettings(
+            0, /* 0 blocks */
+            345600, /* 4 days */
```
Comment (ElliotFriedman) [inline contracts/dao/nopeDAO/NopeDAO.sol:31]: recommend switching to syntax 4 days for clarity

**Comment by thomas-waite** [comment]
Ported over to `feat-governance-upgrade-v2` (git)

File: contracts/pods/PodFactory.sol [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+        external
+        view
+        override
+        returns (address)
+    {
+        return podController.podAdmin(podId);
+    }
+
+    /// @notice Get whether membership transfers are enabled for a pod
+    function getIsMembershipTransferLocked(uint256 podId)
```
Comment (ditchfieldcaleb) [inline contracts/pods/PodFactory.sol:129]: recommend removing "get" from this function name

**Review by Joeysantoro — APPROVED** [review]

**Pull request merged**